quickwit-metastore/migrations/postgresql/4_create-delete_tasks.up.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE IF NOT EXISTS delete_tasks (
+    create_timestamp TIMESTAMP NOT NULL DEFAULT (CURRENT_TIMESTAMP AT TIME ZONE 'UTC'),
+    opstamp BIGSERIAL PRIMARY KEY,
+    index_id VARCHAR(50) NOT NULL,
+    delete_query_json TEXT NOT NULL,
+
+    FOREIGN KEY(index_id) REFERENCES indexes(index_id)
```

Comment:
I think you need to fix `delete_index`.... and possibly have a look at other calls of the metastore I might have not think of.

Right now in presence of delete ops the foreign key will error.
You probably want to delete all delete task in delete_index. 

A unit test would be awesome.

Reply:
Yes indeed.

Reply:
I will use `ON DELETE CASCADE`.

Reply:
We could use `ON DELETE CASCADE`.

Reply:
And there is a test.